pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java

```diff
@@ -240,20 +240,28 @@ public Pair<String, String> extractIdAndDocument(Record<GenericObject> record) t
             if (id != null
                     && idHashingAlgorithm != null
                     && idHashingAlgorithm != ElasticSearchConfig.IdHashingAlgorithm.NONE) {
-                Hasher hasher;
-                switch (idHashingAlgorithm) {
-                    case SHA256:
-                        hasher = Hashing.sha256().newHasher();
-                        break;
-                    case SHA512:
-                        hasher = Hashing.sha512().newHasher();
-                        break;
-                    default:
-                        throw new UnsupportedOperationException("Unsupported IdHashingAlgorithm: "
-                                + idHashingAlgorithm);
+
+                boolean performHashing = true;
+                if (elasticSearchConfig.isConditionalIdHashing()
+                        && id.getBytes(StandardCharsets.UTF_8).length <= 512) {
```

Comment:
do we really need to create the byte[] instance ?
it will generate some garbage

maybe you can create the byte[] here and do not call  `hasher.putString(id, StandardCharsets.UTF_8);` but use the byte[] created here.
I suspect that `putString` will perform the encoding another time

Reply:
good catch!